lms/djangoapps/course_api/serializers.py

```diff
@@ -164,10 +166,19 @@ def to_representation(self, instance):
         """
         Get the `certificate_available_date` in response
         if the `certificates.auto_certificate_generation` waffle switch is enabled
+
+        Get the 'is_enrolled' in response
+        if user is authenticated and 'username' is in query params.
         """
         response = super().to_representation(instance)
         if can_show_certificate_available_date_field(instance):
             response['certificate_available_date'] = instance.certificate_available_date
+
+        requested_user = self.context['request'].query_params.get('username', None)
+        if self.context['request'].user.is_authenticated and requested_user:
```

Comment:
This would let any user look up the enrollment status for any other user, I think we would only want to allow that if the user is a `staff` user and otherwise, we should only provide this info in the requested user is the same as the authenticated user.

Alternatively, we could just flag this on if an authenticated user is making the call and provide the authenticated users enrollment status in that case.  This might be simpler?

Reply:
@feanil The code was updated to follow proposed logic, please review